packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1495,6 +1512,15 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
       }
     }
 
+    if (props.allowAllIpv6Outbound !== undefined) {
+      if (props.securityGroup) {
+        throw new Error('Configure \'allowAllIpv6Outbound\' directly on the supplied SecurityGroup.');
+      }
+      if (hasSecurityGroups) {
+        throw new Error('Configure \'allowAllIpv6Outbound\' directly on the supplied SecurityGroups.');
+      }
```

Comment:
```suggestion
      if (props.securityGroup || hasSecurityGroups) {
        throw new Error('Configure \'allowAllIpv6Outbound\' directly on the supplied SecurityGroup.');
      }
```
Cleaner.

Reply:
Thanks for the suggestion. The difference in the 2 `if` blocks is that the 1st condition is checking if there is a security group specified as part of function props, whereas, in the 2nd condition relies on `hasSecurityGroups` initialized from `props.securityGroups && props.securityGroups.length > 0;`.

This follows the existing pattern for `allowAllOutbound` property.